### PR TITLE
Ignore regex URI when launching

### PR DESCRIPTION
**Pull request by ShirokaiLon**
Closes https://github.com/bitwarden/browser/issues/850 and closes https://github.com/bitwarden/browser/issues/656

### CODE REVIEW

File: src/models/view/loginView.ts [hunk 1]
```diff
@@ -31,11 +31,11 @@ export class LoginView implements View {
     }
 
     get canLaunch(): boolean {
-        return this.hasUris && this.uris[0].canLaunch;
+        return this.hasUris && this.uris.some(uri => uri.canLaunch);
     }
 
     get launchUri(): string {
-        return this.canLaunch ? this.uris[0].launchUri : null;
+        return this.canLaunch ? this.uris.find(uri => uri.canLaunch).launchUri : null;
```
[the remark this answers is not in the document]
Comment (ShirokaiLon) [reply src/models/view/loginView.ts]: Woops, wrong check. Fixed it.

File: src/models/view/loginView.ts [hunk 2]
```diff
@@ -31,11 +31,11 @@ export class LoginView implements View {
     }
 
     get canLaunch(): boolean {
-        return this.hasUris && this.uris[0].canLaunch;
+        return this.hasUris && this.uris.some(uri => uri.canLaunch);
     }
 
     get launchUri(): string {
-        return this.canLaunch ? this.uris[0].launchUri : null;
+        return this.hasUris ? this.uris.find(uri => uri.canLaunch).launchUri : null;
```
Comment (kspearrin) [inline src/models/view/loginView.ts]: Couldn't this lead to a null ref if find returns null here? seems you need to store find value in a variable and null check it

Comment (ShirokaiLon) [reply src/models/view/loginView.ts]: Yes... :disappointed_relieved:

**Comment by kspearrin** [comment]
@ShirokaiLon Thanks. For future PRs, please run the linter (`npm run lint`) and correct any issues listed there. ref: https://github.com/bitwarden/jslib/commit/98addd8ab5a226f654e78b1fe806b5fab6575b68

**Pull request merged**